python/lsst/ts/ofc/controllers/base_controller.py

```diff
@@ -250,7 +270,17 @@ def pid_step(self, state: np.ndarray) -> np.ndarray:
         self.integral += error
         derivative = error - self.previous_error
 
-        uk = self.kp * error + self.ki * self.integral + self.kd * derivative
+        # Apply low-pass filter to the derivative term
+        self.filtered_derivative = (
+            self.derivative_filter_coeff * derivative
+            + (1 - self.derivative_filter_coeff) * self.filtered_derivative
+        )
```

Comment:
The current implementation is a very simple one. In the future, we might need to abstract this part to a new class to let the users can experiment different filters. In the field of control, sometimes, people would say the control itself is the tuning of filters. Actually, in the frequency domain, you could see the PI controller is a low-pass filter and the PD controller is a high-pass filters as well.